pages/index.js

```diff
@@ -20,47 +19,38 @@ export async function getStaticProps() {
     price: '',
   }));
 
-  productsWithPrices.forEach((element, i) => {
-
-    // 1. this iterates over each element once
-    // 2. for each element, compare its id value to every
-    // product value within the prices array looking for a match
-    // 3. if matched, take the unit_amount from the matching price object
-    // and ascribe that value to the price item in the active product element
-    // if (element[i].id === prices[i].products) {
-    // eslint-disable-next-line no-console
-    // const matchingPrice = prices.includes(element.id);
-    // console.log(element.id);
-    if (element.id !== prices[i].product) {
-      //   // element.price = prices[i].unit_amount;
-      console.log(prices[i].product, element.id);
+  function integratePrice() {
+    for (let i = 0; i < productsWithPrices.length; i++) {
+      const matchingPrice = prices.filter((price) => price.product === productsWithPrices[i].id);
+      if (matchingPrice[0] === undefined) {
+        matchingPrice[0] = { unit_amount: null };
+      }
+      productsWithPrices[i].price = matchingPrice[0].unit_amount;
     }
-    // }
-  });
+  }
+  integratePrice();
```

Comment:
ditto, we could make this a reusable helper function if we're gonna use it more than once.